### ISSUE TITLE
How to deal with anonymous imports?

### ISSUE THREAD

**Issue by cowwoc** [rollup/rollup#1217]
https://babeljs.io/docs/usage/polyfill/#usage-in-browser recommends the use of an anonymous import statement:

  `import "babel-polyfill"`

But if I do this, rollup warns:

```
Treating 'babel-polyfill' as external dependency
No name was provided for external module 'babel-polyfill' in options.globals – guessing 'babelPolyfill'
```

Further, when I try using the resulting bundle in a browser, I get this error:

```
Uncaught ReferenceError: babelPolyfill is not defined
```

1. How are we supposed to handle shims that augment existing globals without necessarily exporting their own?
2. Is there a way to suppress rollup's warnings?
3. I'm not sure that the output bundle should be waiting on `babelPolyfill` since anonymous imports imply that a global might not be getting exported.

**Comment by cowwoc** [comment]
As an aside, in the specific case of `babel-polyfill` I discovered that it exports an undocumented global `_babelPolyfill` so I used:

```
external: ["babel-polyfill"],
globals: {
	"babel-polyfill": "_babelPolyfill"
}
```

to suppress rollup's warnings. But I'm still curious what the best-practice is for these kinds of situation (in case I run into the problem again with a different library).

**Comment by Rich-Harris** [comment]
`import 'babel-polyfill'` means 'this bundle depends on babel-polyfill'. If Rollup knows how to find `babel-polyfill`, it will include it in your bundle. But by default, it *doesn't* know, so you get a message about it being treated as an external dependency. To tell it how to find it in `node_modules` you need to use [rollup-plugin-node-resolve](https://github.com/rollup/rollup-plugin-node-resolve).

Rollup should probably ignore missing globals for empty imports, come to think of it. Will mark this as a bug for that reason.

**Comment by cowwoc** [comment]
Thanks excellent. Thanks Rich!

**Comment by cowwoc** [comment]
Odd. I just tried version 0.40.2 and I still get the same error.

Are you able to reproduce this on your end, or do you need me to provide a testcase for it?

**Comment by cowwoc** [comment]
Correction, the error changed. Now I get:

```
'babel-polyfill' is imported by src\Requirements.js, but could not be resolved – treating it as an external dependency. For help see https://github.com/rollup/rollup/wiki/Troubleshooting#treating-module-as-external-dependency
No name was provided for external module 'babel-polyfill' in options.globals – guessing 'babelPolyfill'
```

**Comment by Rich-Harris** [comment]
That's strange. Can you provide a repro? (you'll still need to use the node-resolve plugin as per my earlier comment, but it shouldn't print the `options.globals` warning)

**Comment by cowwoc** [comment]
@Rich-Harris Sorry for the false alarm. I had totally forgotten about the node-resolve plugin. I confirm that this issue is fixed in 0.40.2. Thank you for the quick turnaround!
